contracts/Pricing.sol

```diff
@@ -76,33 +76,42 @@ contract Pricing is IPricing {
      */
     function recordTrade(uint256 tradePrice, uint256 fillAmount) external override onlyTracer {
         uint256 currentOraclePrice = oracle.latestAnswer();
+        // update rates if a trade has not been recorded in the last hour
         if (startLastHour <= block.timestamp - 1 hours) {
-            // emit the old hourly average
+            // get the last recorded hourly price, returns max integer if no trades occurred
             uint256 hourlyTracerPrice = getHourlyAvgTracerPrice(currentHour);
-            emit HourlyPriceUpdated(hourlyTracerPrice, currentHour);
 
-            // update funding rate for the previous hour
-            updateFundingRate();
+            // emit the hourly average and udpate funding rate if trades occurred
+            if (hourlyTracerPrice != type(uint256).max) {
+                // emit the old hourly average
+                emit HourlyPriceUpdated(hourlyTracerPrice, currentHour);
+
+                // update funding rate for the previous hour
+                updateFundingRate();
+            }
+
+            uint256 elapsedHours = (block.timestamp - startLastHour) / 3600;
```

Comment:
This should be capped at 24 (ternary works) -- for a market that hasn't been active for a few weeks/months (however unlikely), this would go into a very long loop